src/main/scala/mesosphere/marathon/api/v2/QueueResource.scala

```diff
@@ -38,16 +42,35 @@ class QueueResource @Inject() (
     ok(Raml.toRaml((stats, embedLastUnusedOffers, clock)))
   }
 
+  @GET
+  @Path("""{runSpecId:.+}/delay""")
+  def getDelay(
+    @PathParam("runSpecId") id: String,
+    @Context req: HttpServletRequest): Response = authenticated(req) { implicit identity =>
+    val runSpecId = id.toRootPath
+    withAuthorization(
+      ViewRunSpec,
+      groupManager.runSpec(runSpecId),
+      notFound(appNotFoundTasksQueue(runSpecId))) { app =>
+        val queueDelay = result(launchStats.getStatistics())
+          .find(_.runSpec.id equals runSpecId)
+          .flatMap(_.queueDelay(clock))
+        result(launchQueue.getDelay(app)).delay match {
+          case Some(delay) => ok(DelayResult(queueDelay, delay.currentDelay.toSeconds, delay.maxLaunchDelay.toSeconds))
+          case None => notFound(delayNotFound(runSpecId))
+        }
+      }
+  }
+
   @DELETE
-  @Path("""{appId:.+}/delay""")
+  @Path("""{runSpecId:.+}/delay""")
   def resetDelay(
-    @PathParam("appId") id: String,
+    @PathParam("runSpecId") id: String,
     @Context req: HttpServletRequest): Response = authenticated(req) { implicit identity =>
-    //    import scala.concurrent.ExecutionContext.Implicits.global
-    val appId = id.toRootPath
-    val appScheduled = result(instanceTracker.specInstances(appId)).exists(_.isScheduled)
-    val maybeApp = if (appScheduled) groupManager.runSpec(appId) else None
-    withAuthorization(UpdateRunSpec, maybeApp, notFound(s"Application $appId not found in tasks queue.")) { app =>
+    val runSpecId = id.toRootPath
+    val appScheduled = result(instanceTracker.specInstances(runSpecId)).exists(_.isScheduled)
```

Comment:
I love updating the nomenclature here; I'd also like to replace appScheduled with runSpecScheduled, maybeApp with maybeRunSpec, app with runSpec, etc.

Reply:
Related, it seems this PR is starting to scope creep. We'd generally prefer to keep our PRs smaller as it is easier to review that way. Realizing this is a WIP PR, when you're ready to submit, might it be possible to just do the show delay in one PR, and then the extra delete changes in another?

Reply:
It wasn't intentional to creep the scope of the PR but i agree that this got bigger than what I intended. There are no behavioral changes related to `DELETE /v2/queue/<id>/delay` endpoint. So far, everything has been related to `GET` endpoint and some other cleanup I did as I was reading/modifying code (i ll be more conscious editing unrelated code in future). I am not sure  if splitting the PR makes sense now as it will only be a formatting fix.